typescript-requiring-type-checking.js

```diff
@@ -15,23 +15,23 @@ module.exports = {
         /*
```

Comment:
Discussion for recommended rule: `'@typescript-eslint/no-for-in-array': 'error'`

Reply:
This case is specifically preventing use of for-in for Arrays and not Objects. My understanding of:
> For...in statements can be used when performance is a consideration as long as guard-for-in is enable

Was for performance considerations when accessing members of an object. I don't think we should ever use for..in for iterating across indexes of an array, particularly if we are citing performance as the reason.

When targeting performance arrays should be iterated using three-clause for-loops `for (let i = 0; i < arr.length; i++) {}`

Proposal: Do not disable the recommended rule

Reply:
We agreed on Milan's proposal in person.